src/actions/RegistrationActions.js

```diff
@@ -0,0 +1,55 @@
+import toastr from "../helpers/toastrConfig";
+
+import { REGISTER_REQUEST, REGISTER_SUCCESS, REGISTER_FAILURE } from "./actionTypes";
+import instance from "../api/axiosConfig";
+
+/* eslint-disable import/prefer-default-export */
+export const registerRequest = () => ({
+  type: REGISTER_REQUEST,
+  isFetching: true
+});
+
+export const registerSuccess = response => ({
+  type: REGISTER_SUCCESS,
+  isFetching: false,
+  response
+});
+
+export const registerFailure = errors => ({
+  type: REGISTER_FAILURE,
+  isFetching: false,
+  errors
+});
+
+const registrationAsync = user => (dispatch) => {
+  dispatch(registerRequest());
+  instance
```

Comment:
Can you move the API call to an API class method, then dispatch actions according to what has been returned from API class method?